Userland/DevTools/Profiler/FlameGraphView.cpp

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright (c) 2021, Nicholas Hollett <niax@niax.co.uk>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "FlameGraphView.h"
+#include "DevTools/Profiler/Profile.h"
+#include "LibGfx/Forward.h"
+#include <AK/Function.h>
+#include <LibGUI/Painter.h>
+#include <LibGfx/FontDatabase.h>
+#include <LibGfx/Palette.h>
+
+namespace Profiler {
+
+constexpr int bar_rounding = 2;
+constexpr int bar_margin = 2;
+constexpr int bar_padding = 8;
+constexpr int bar_height = 20;
+constexpr int text_threshold = 30;
+
+Vector<Gfx::Color> s_colors;
+
+static Vector<Gfx::Color> const& get_colors()
+{
+    if (s_colors.size() == 0) {
+        // Start with a nice orange, then make shades of it
+        Gfx::Color midpoint(255, 94, 19);
+        s_colors.append(midpoint);
+        for (auto i = 0.1f; i < 0.6f; i += 0.2f) {
+            s_colors.append(midpoint.lightened(1.f + i));
+            s_colors.append(midpoint.darkened(1.f - i));
+        }
+    }
+
+    return s_colors;
+}
```

Comment:
Here's another implementation of something similar, from `Userland/DevTools/Profiler/DisassemblyModel.cpp`:

```cpp
static const Gfx::Bitmap& heat_gradient()
{
    static RefPtr<Gfx::Bitmap> bitmap;
    if (!bitmap) {
        bitmap = Gfx::Bitmap::try_create(Gfx::BitmapFormat::BGRx8888, { 101, 1 });
        GUI::Painter painter(*bitmap);
        painter.fill_rect_with_gradient(Orientation::Horizontal, bitmap->rect(), Color::from_rgb(0xffc080), Color::from_rgb(0xff3000));
    }
    return *bitmap;
}

static Color color_for_percent(int percent)
{
    VERIFY(percent >= 0 && percent <= 100);
    return heat_gradient().get_pixel(percent, 0);
}
```

Your implementation seems to yield nicer colors, so it would be nice for everyone to share it. Maybe this could be a helper in LibGUI? :)

Reply:
I've added `shades` and `tints` to `Gfx::Color`, I can update `DisassemblyModel` in a future PR.